### PR TITLE
Add functions commonly used by custom WA authors

### PR DESCRIPTION
**Pull request by asakawa-k**
WeakAuras.UnitAura, WeakAuras.UnitBuff, WeakAuras.UnitDebuff - mimicking functionality from pre-8.0
WeakAuras.GroupMembers - Very popular in Legion.

This would be a bit of a step off the normal path so I totally understand if providing stuff for custom WA authors isn't something you want to get into, or if the functions I added simply aren't strong enough that you want them to be carried by your addon. 
However, I thought that if I throw a PR up then it at least sparks the discussion and you can always reject the PR if you're not keen.

### CODE REVIEW

File: WeakAuras/WeakAuras.lua [hunk 1]
```diff
@@ -4756,3 +4756,37 @@ function WeakAuras.PrintProfile()
   end
   print("--------------------------------");
 end
+
+-- Unit Aura functions that return info about the first Aura matching the spellName or spellID given on the unit. 
+function WeakAuras.UnitAura(unit, spell, filter)
```
Comment (InfusOnWoW) [inline WeakAuras/WeakAuras.lua]: Apparently you wanted local function GetUnitAura here.

**Comment by InfusOnWoW** [comment]
I think I prefer to keep them in a separate namespace, as not everything in WeakAuras should be called by the user. (Or actually most shouldn't.)

**Comment by Stanzilla** [comment]
I'd probably want them in an extra file.

File: WeakAuras/HelperFunctions.lua [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          local name, _, _, _, _, _, _, _, _, spellId = UnitAura(unit, i, filter)
+          if not name then return end
+          if spell == spellId or spell == name then
+              return UnitAura(unit, i, filter)
+          end
+      end
+  end
+  ,
+  GetUnitBuff = function(unit, spell, filter)
+      return GetUnitAura(unit, spell, filter)
```
Comment (emptyrivers) [inline WeakAuras/HelperFunctions.lua]: Ensure that the filter either doesn't exist or contains HELPFUL. Also, GetUnitAura here refers to a nonexistent global. Instead, use `return WeakAuras.helperFunction.GetUnitAura`.

File: WeakAuras/HelperFunctions.lua [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  end
+  ,
+  GetUnitDebuff = function(unit, spell, filter)
+      filter = filter and filter.."|HARMFUL" or "HARMFUL"
+      return GetUnitAura(unit, spell, filter)
```
Comment (emptyrivers) [inline WeakAuras/HelperFunctions.lua]: GetUnitAura is a nonexistent global here as before

File: WeakAuras/HelperFunctions.lua [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+          end
+      end
+  end
+  ,
+  GetUnitBuff = function(unit, spell, filter)
+      return GetUnitAura(unit, spell, filter)
+  end
+  ,
+  GetUnitDebuff = function(unit, spell, filter)
+      filter = filter and filter.."|HARMFUL" or "HARMFUL"
```
Comment (emptyrivers) [inline WeakAuras/HelperFunctions.lua]: Note that this is vulnerable to someone going `GetUnitDebuff("player", "some spell name",  "HELPFUL")`. I have no idea if this is worth guarding against, or if it should just be garbage in garbage out.

**Comment by emptyrivers** [comment]
It may well be worth moving all of the `aura_env` support code into `HelperFunctions.lua` at some point, in part of the ongoing battle to keep these files at a manageable size.

**Comment by InfusOnWoW** [comment]
As TheDanW confirmed that he'll add something to wow after all, we'll wait a few more days to see what exactly they added.

**Comment by asakawa-k** [comment]
Yeah. Fingers tightly crossed. 

If they do put the functionality back in to UnitAura, I'd like to keep the discussion alive regarding adding useful functions for custom WA authors. But at that point it will be significantly less important/urgent and, if you were open to it, there may be a better way to implement it.

**Comment by asakawa-k** [comment]
Yeah, that's the kind of thing that could be done properly if there isn't the urgency of the pre-patch requiring something in place.

**Comment by InfusOnWoW** [comment]
Can you squash all the commits? I think something went wrong and you duplicated a commit or so.

**Comment by InfusOnWoW** [comment]
Thanks.

**Pull request merged**